src/screens/ImagesGallery.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useMemo } from "react";
+import {
+  Text,
+  StyleSheet,
+  View,
+  Image,
+  TouchableOpacity,
+  FlatList,
+} from "react-native";
+
+import { SafeAreaView } from "react-native-safe-area-context";
+
+import Menu from "../components/Menu";
+import typography from "../assets/theme/typography";
+import InfoModal from "../components/InfoModal";
+import BottomNavButtons from "../components/BottomNavButtons";
+import { useActivityImages } from "../context/ActivityImagesContext/ActivityImagesContext";
+import { ActivityImage } from "../hooks/useCloudImages";
+import { selectedImageStyle } from "../styles/selectedImageStyle";
+
+enum Routes {
+  CreateActivity = "CreateActivity",
+  CreateUser = "CreateUser",
+  ChangeActivity = "ChangeActivity",
+}
+
+export function ImagesGallery({
+  navigation,
+  route,
+}: {
+  navigation: any;
+  route: any;
+}) {
+  const { selectedImage, cameFrom, tgPopular, activity } = route.params;
+  const { images } = useActivityImages();
+
+  const [selected, setSelected] = useState<ActivityImage>(
+    selectedImage ?? { photo: "" },
+  );
+
+  const imagesArray = useMemo(() => {
+    return images
+      .map((img) => ({
+        ...img,
+        selected: img.photo === selected.photo,
+      }))
+      .sort((a, b) => a.photo.localeCompare(b.photo));
+  }, [images, selected]);
+
+  const buttonSavePressed = () => {
+    const routeMapping = {
```

Comment:
Just export that Routes enum and use it at places you need it.  You don't need to make an object that is just a copy of the Routes enum and  send as type Routes